src/engine/ui/widgets/Image.cxx

```diff
@@ -9,3 +10,15 @@ void Image::draw()
     SDL_SetTextureAlphaMod(m_texture, SDL_ALPHA_OPAQUE);
   }
 }
+
+void Image::setOpacity(Uint8 alphaLevel)
+{
+  if (0 <= alphaLevel <= 255)
```

Comment:
if `alphaLevel` is a uint8, this will always evaluate to true. Moreover, this does not actually check if `alphaLevel` is smaller than 255, because if `alphaLevel` was an int32, and was equal to 300, then `0 <= alphaLevel` would be `true` which is converted to `1`, and then `1 <= 255` would always be true. So in C++ `0 <= 300<= 255` is true because of left associativity. If we really needed this check, then we would need to do `0 <= alphaLevel && alphaLevel <= 255`